test/jdk/java/lang/Character/Supplementary.java

```diff
@@ -797,12 +797,15 @@ private static void callCodePoint(boolean isAt, char[] a, int index,
     }
 
     private static void callCodePoint(boolean isAt, char[] a, int index, int limit,
-                                      Class expectedException) {
+                                      Class<? extends Exception> expectedException) {
         try {
             int c = isAt ? Character.codePointAt(a, index, limit)
                          : Character.codePointBefore(a, index, limit);
         } catch (Exception e) {
-            if (expectedException.isInstance(e)) {
+            if (isAt && expectedException == e.getClass()) {
+                return;
+            }
+            if (expectedException.isInstance(e)) { // Character.codePointBefore() throws
```

Comment:
The test should fail without the fix, but this fall-through allows the test to pass.

The qualification added for `isAt` is hiding a second/corresponding bug in `codePointBefore` where the index is not checked and an ArrayIndexOutOfBounds occurs at codePointBeforeImpl: 9488.

Reply:
Yeah, this AIOOBE is exactly the reason why I added this. So should we modify `codePointBefore` in the same way as `codePointAt`?

Reply:
I'm asking because counter-intuitively `codePointBefore ` doesn't specify IOOBE for negative `index`.
```
/**
 * @return the Unicode code point value before the given index.
 * @throws NullPointerException if {@code a} is null.
 * @throws IndexOutOfBoundsException if the {@code index}
 * argument is not greater than the {@code start} argument or
 * is greater than the length of the {@code char} array, or
 * if the {@code start} argument is negative or not less than
 * the length of the {@code char} array.
 * @since  1.5
 */
 ```